app/models/publication.rb

```diff
@@ -56,10 +56,6 @@ def self.find_or_create_by_pmid(pmid)
     find_by_pmid(pmid) || SciencewireSourceRecord.get_pub_by_pmid(pmid) || PubmedSourceRecord.get_pub_by_pmid(pmid)
   end
 
-  def self.find_or_create_by_sciencewire_id(sw_id)
-    find_by_sciencewire_id(sw_id) || SciencewireSourceRecord.get_pub_by_sciencewire_id(sw_id)
-  end
-
```

Comment:
`Publication` has no method defined for `'find_by_sciencewire_id`, i.e.

```
$ git grep 'def find_by_sciencewire_id'  # returns nothing
$ git grep 'find_by_sciencewire_id' app/models/
app/models/publication.rb:60:    find_by_sciencewire_id(sw_id) || SciencewireSourceRecord.get_pub_by_sciencewire_id(sw_id)
app/models/publication.rb:147:      sw_source_record = SciencewireSourceRecord.find_by_sciencewire_id(sciencewire_id)
```

So if this method works, it must be an AR dynamic method; wondering what rubocop is complaining about that needs to be silenced?  There is a db-field for `Publication#sciencewire_id`, perhaps this rubocop warning is to avoid the dynamic method and use the named field, e.g. maybe it should be this instead:
```
Publication.find_by(sciencewire_id: sw_id)
```

Reply:
Ah, that is a good point.  Might be able to satisfy rubocop after all.

Reply:
Yeah, that worked after all.